ppl/archive/import.go

```diff
@@ -91,11 +113,50 @@ func (w *Writer) Write(rec *zng.Record) error {
 	return nil
 }
 
+// flusher is a background go-routine for an active Writer that periodically
+// checks for tsdir writers that have not recieved data in a while. If such a
+// writer is found, it is flushed to disk and closed.
+func (w *Writer) flusher() error {
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			if err := w.flushStaleWriters(); err != nil {
+				return err
+			}
+		case <-w.ctx.Done():
+			return nil
+		}
+	}
+}
+
+func (w *Writer) flushStaleWriters() error {
+	w.mu.Lock()
+	var stale []*tsDirWriter
+	now := time.Now()
```

Comment:
No need for the lock across time.Now.
```suggestion
	now := time.Now()
	var stale []*tsDirWriter
	w.mu.Lock()
```